consensus/src/main/kotlin/maru/consensus/BlockCreator.kt

```diff
@@ -0,0 +1,136 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.consensus
+
+import maru.core.BeaconBlock
+import maru.core.BeaconBlockBody
+import maru.core.BeaconBlockHeader
+import maru.core.BeaconState
+import maru.core.HashType
+import maru.core.HashUtil
+import maru.core.Seal
+import maru.core.Validator
+import maru.database.BeaconChain
+import maru.executionlayer.manager.ExecutionLayerManager
+import org.apache.logging.log4j.LogManager
+import org.hyperledger.besu.consensus.common.bft.ConsensusRoundIdentifier
+import org.hyperledger.besu.consensus.common.bft.blockcreation.ProposerSelector
+import org.hyperledger.besu.consensus.qbft.core.types.QbftBlockCreator
+
+/**
+ * Responsible for beacon block creation.
+ */
+class BlockCreator(
+  private val manager: ExecutionLayerManager,
+  private val proposerSelector: ProposerSelector,
+  private val validatorProvider: ValidatorProvider,
+  private val beaconChain: BeaconChain,
+) {
+  private val log = LogManager.getLogger(QbftBlockCreator::class.java)
+
+  /**
+   * Creates a new block with the given timestamp on of the parent header including the execution payload ready
+   * to be proposed as a block in the QBFT consensus.
+   */
+  fun createBlock(
+    timestamp: Long,
+    roundNumber: Int,
+    parentHeader: BeaconBlockHeader,
+  ): BeaconBlock? {
```

Comment:
Can we avoid nullability here? Would it be appropriate if we throw if `finishBlockBuilding` doesn't build anything?